app/models/page.rb

```diff
@@ -4,7 +4,11 @@ class Page < ApplicationRecord
   validates :title,       presence: true
   validates :slug,        presence: true
   validates :template_id, presence: true
-  validates :hidden,      inclusion: { in: [ true, false ] }
+  validates :hidden,      inclusion:  { in: [ true, false ] }
+  validates :slug,        uniqueness: {
```

Comment:
Consider adding a corresponding composite, unique index on the slug and section_id table columns, to enforce this constraint.